File: C/10828.c
#include <stdio.h>  // Ω∫≈√
#include <string.h>

int stack[10000];
int count = 0;

void push(int num)
{
	stack[count] = num;
	count++;
}

void pop()
{
	if (count != 0)
	{
		count--;
		printf("%d\n", stack[count]);
	}
	else
		printf("%d\n", -1);
}

void size()
{
	printf("%d\n", count);
}

void empty()
{
	if (count != 0)
		printf("%d\n", 0);
	else
		printf("%d\n", 1);
}

void top()
{
	if (count != 0)
		printf("%d\n", stack[count - 1]);
	else
		printf("%d\n", -1);
}


int main()
{
	int n;
	scanf("%d", &n);

	char str[10];
	for (int i = 0; i < n; i++)
	{
		scanf("%s", str);
		if (strcmp(str, "push") == 0)
		{
			int num;
			scanf("%d", &num);
			push(num);
		}
		else if (strcmp(str, "pop") == 0)
			pop();
		else if (strcmp(str, "size") == 0)
			size();
		else if (strcmp(str, "empty") == 0)
			empty();
		else if (strcmp(str, "top") == 0)
			top();
	}

	return 0;
}

File: Python/10828.py
## 스택
import sys
input = sys.stdin.readline

n = int(input())

arr = []
for _ in range(n):
    command = input().split()
    if command[0] == "push":
        arr.append(int(command[1]))
    elif command[0] == "pop":
        if len(arr) == 0:
            print(-1)
        else:
            print(arr.pop())
    elif command[0] == "size":
        print(len(arr))
    elif command[0] == "empty":
        if len(arr) == 0:
            print(1)
        else:
            print(0)
    elif command[0] == "top":
        if len(arr) == 0:
            print(-1)
        else:
            print(arr[-1])